### PR TITLE
⚡️ schema generation performance

### PR DESCRIPTION
**Pull request by heikomat**
**TL;DR**: I removed unused, and deduped heavy typescript-function-calls. Please check if you can find anything that you think could break our current schema-generation

**Long form**
- I removed every occurrence of `ts.getPreEmitDiagnostics(program);`. I don't know what this line does (as its return-value was always ignored), it was pretty cpu-intensive, and things still seem to work without it.
  - As a result i could remove a now unused `ts.createProgramm(...)` call
- I removed `const indexProgram = ts.createProgram(...)` because `indexProgram` wasn't used anywhere
- the compilerHost was created three times with identical parameters. I moved code around, so it only gets created once
- The code from creating the compilerHost to getting the entry-exports was identical in the functions `getExportedSymbols` and `getExportHeritage` and was also very time consuming, so i deduped it by moving it out of the functions.

All these changes shorten the generation of all schemas for the process-engine on my laptop from 13:35 minutes to 7:32 minutes, a 44% improvement. This shortens the total meta-setup-time on my laptop to just over 21 minutes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     "del": "^3.0.0",
     "fs-extra": "^4.0.0",
     "glob": "^7.1.2",
+    "gulp": "^3.9.1",
```
Comment (5ebastianMeier) [inline package.json]: A gulptraum plugin shall never define `gulp` as its dependency.
It should be a peer dependency or none at all.

If there is a specific reason for the `gulp` dependency, we need to isolate it as much as possible.

Gulptraum is able to run on gulp@3, as well as gulp@4 and I want to keep it that way.
